src/main/java/redis/clients/jedis/util/JedisURIHelper.java

```diff
@@ -74,7 +78,7 @@ public static RedisProtocol getRedisProtocol(URI uri) {
 
   private static final Integer ZERO_INTEGER = 0;
 
-  public static ClientSideCache getClientSideCache(URI uri) {
+  public static ClientSideCacheConfig getClientSideCache(URI uri) {
```

Comment:
Maybe add a new argument, a cache config builder? Can be `Function<ClientSideCache, ClientSideCacheConfig>`. That might be a breaking change, so it can be a new method, to have an alternative.